doc/changelog.txt

```diff
@@ -32,6 +32,63 @@ BREAKING CHANGES & HOW TO GET BACK OLD BEHAVIOURS:
      return originalGetConfigInt(key, def) or 0
    end
 
+ ! hovercraft and ships brought out of water no longer forced to be upright.
+   To get back previous behaviour, put the `upright = true` tag in all unit defs
+   whose move def is of the hovercraft or ship type.
+
+ ! The return value from the UnitUnitCollision callin is now ignored
+   and there is only one event for each collision.
+   There is no way to get back the old behaviour for now,
+   but if someone needs it it could be arranged.
+
+ ! The GameProgress callin is now called every 5 seconds (from 10).
+   This value is not exposed anywhere so to get back the previous
+   behaviour you will have to find whichever code assumes the old value.
+
+ ! Buildings can be placed inside other buildings' open yardmap gaps.
+   Use the AllowCommand or AllowUnitCreation callins to prevent this.
+
+ ! Rendering fonts now obeys GL color state. To get back previous
+   behaviour, add gl.Color in front of font calls.
```

Comment:
This would ideally be double-checked. Was the color always white previously (before any `\0\255\255` style color characters)? There seems to be no getter for color so you can't do
```lua
local r, g, b, a = gl.GetCurrentColor() -- doesn't exist
gl.Color(1,1,1,1) -- was it previously white?
-- print stuff
gl.Color(r,g,b,a)
```

Reply:
The interaction is complicated there, it used to rely on GL state, including state stored in display lists. `glColor` is no longer stored as part of the rendering primitive, so relying on the color of text became slightly flaky. Now the color is stored as global variable, which in theory should at least guarantee that if in cases the color is not what it should be, setting `glColor` would force the right front color. Despite all that it's still not guaranteed that pre-Recoil game moving here, would get exact same behavior that it used to have, so sometimes new `glColor` before `gl.Text` will be necessary.